includes/Media.php

```diff
@@ -273,4 +277,45 @@ public static function upload_mimes( array $mime_types ) {
 		$mime_types['svg'] = 'image/svg+xml';
 		return $mime_types;
 	}
+
+	/**
+	 * Deletes associated poster image when a video is deleted.
+	 *
+	 * This prevents the poster image from becoming an orphan because it is not
+	 * displayed anywhere in WordPress or the story editor.
+	 *
+	 * @param int $attachment_id ID of the attachment to be deleted.
+	 *
+	 * @return void
+	 */
+	public static function delete_video_poster( $attachment_id ) {
+		remove_action( 'pre_get_posts', [ __CLASS__, 'filter_poster_attachments' ] );
+		$query = new WP_Query(
+			[
+				'fields'         => 'ids',
+				'post_status'    => 'any',
+				'post_type'      => 'attachment',
+				'post_parent'    => $attachment_id,
+				// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_query
```

Comment:
Let's above the meta query here. It is not needed, just get all the attachments with the parent id and to do a `get_post_meta`. meta_queries are slow and should be avoided.